mathinspector/__init__.py

```diff
@@ -1,2 +1,217 @@
+		self.objects = ObjectTree(self)
+
+		self.vertical_panel.add(self.node.frame)
+		self.vertical_panel.add(self.console.frame)
+		self.side_view.add("Modules", self.modules)
+		self.side_view.add("Objects", self.objects)
+		self.horizontal_panel.add(self.side_view.frame)
+		self.horizontal_panel.add(self.vertical_panel)
+		self.horizontal_panel.pack(side="left", fill="both", expand=True)
+
+		self.animate = Animation(self)
+		self.project = SaveData(self)
+		self.menu = MainMenu(self)
+		self.config(menu=self.menu)
+
+		self.title("Math Inspector")
+		self.debug = debug
+
+		mathfile = AUTOSAVE_PATH
+		pyfiles = []
+		for i in args:
+			if not os.path.isfile(i):
+				builtin_print("mathinspector failed to launch: " + i + " is not a file\n")
+				exit()
+
+			name, ext = name_ext(i)
+			if ext == ".math":
+				mathfile = os.path.abspath(i)
+			elif ext == ".py":
+				mathfile = ""
+				pyfiles.append(os.path.abspath(i))
+			else:
+				builtin_print("mathinspector failed to launch: " + ext + " is not a currently supported file type.\n")
+				exit()
+
+		self.project.load(mathfile, is_first_load=True, sashpos=240 if pyfiles else 0)
+
+		for i in pyfiles:
+			self.modules.addfile(i)
+
+
+def main(*args, **kwargs):
+	"""
+	launches the mathinspector app
+
+	Parameters
+	----------
+	*args : This function accepts an arbitrary number of arguments
+		Each argument must be a filename in the relative path which has at most a single .math file and can have an arbitrary number of .py files.  When arguments are passed to this function, the app will launch with the .math file loaded and all of the .py files added
+
+	help: string (optional)
+		launches the mathinspector documentation system as if the command help(STRING) was called from the mathinspector app, where STRING is the value of the help kwarg.
+
+	new: bool
+		When set to True, this will replace the current autosave file with a blank file and when the app launches it will be the same thing as if you selecte File > New from the main menu.
+
+	debug: bool
+		When set to True, this will print a range of log messages to the command line used to launch the app.  Useful for debugging issues or while working on the math inspector source code.
+
+
+	Examples
+	--------
+
+	launch the app with a brand new project
+	>>> mathinspector.main(new=True)
+
+	launch the app and add test.py to the project
+	>>> mathinspector.main("~/Projects/test.py")
+
+	launch the app with the project stored in myproject.math open
+	>>> mathinspector.main(mathfile="myproject.math")
+
+	launch the app in debug mode
+	>>> mathinspector.main(debug=True)
+
+	"""
+	if "help" in kwargs:
+		return help(kwargs["help"])
+
+	if "new" in kwargs and kwargs["new"] is True:
+		with open(AUTOSAVE_PATH, "w") as f:
+			f.write("")
+			f.close()
+
+	params = {
+		"debug": kwargs["debug"] if "debug" in kwargs else False
+	}
+
+	app = App(*args, **params)
+	sys.stderr = StdWrap(sys.stderr, app.console.write) # overrides stderr after init app
```

Comment:
why in a different place from `stdout` wrapping?

Reply:
When sys.stderr is in Interpreter, and there is a syntax error in the math inspector source code, the app will fail to launch and there are no error messages.

So I moved it into `main` to be sure to be able to see error messages when that happens, and left the other stuff where it is while trying to figure out if all override code should actually be in main.